src/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -1082,11 +1085,52 @@ QgsRasterIdentifyResult QgsGdalProvider::identify( const QgsPointXY &point, QgsR
       else
         results.insert( i, value );
     }
-    delete myBlock;
   }
   return QgsRasterIdentifyResult( QgsRaster::IdentifyFormatValue, results );
 }
 
+bool QgsGdalProvider::worldToPixel( double x, double y, int &col, int &row ) const
+{
+  double div = ( mGeoTransform[2] * mGeoTransform[4] - mGeoTransform[1] * mGeoTransform[5] );
+  if ( div < 2 * std::numeric_limits<double>::epsilon() )
+    return false;
+  double doubleCol = -( mGeoTransform[2] * ( mGeoTransform[3] - y ) + mGeoTransform[5] * x - mGeoTransform[0] * mGeoTransform[5] ) / div;
+  double doubleRow = ( mGeoTransform[1] * ( mGeoTransform[3] - y ) + mGeoTransform[4] * x - mGeoTransform[0] * mGeoTransform[4] ) / div;
+  col = static_cast< int >( doubleCol );
+  row = static_cast< int >( doubleRow );
```

Comment:
Is it correct to round down here? (I don't know, just asking...)

Reply:
Yes, it's correct - otherwise the values are 0.5 pixels off